File: api/src/main/java/fr/phenix333/awale/arena/model/Bot.java
package fr.phenix333.awale.arena.model;

import java.util.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
@Table(name = "bots")
public class Bot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "created_at")
    private Date createdAt = new Date();

    @Column(name = "commit_link")
    private String commitLink;

    @Column(nullable = false)
    private String name;

    @ManyToOne
    @JoinColumn(name = "\"group\"", nullable = false)
    private Group group;

    private boolean activated = true;

}

File: interface/app/types/bot.ts
import type { Group } from "@/types/group";

export interface SupabaseBot {
    id: number;
    created_at: string;
    commit_link: string;
    name: string;
    group: number;
    activated: boolean;
}

export interface Bot {
    id: number;
    createdAt: Date;
    commitLink: string;
    name: string;
    group: Group;
    activated: boolean;
}

export interface AddBotPayload {
    name: string;
    commitLink: string;
    group: { name: string };
}
